### PR TITLE
Explore page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Explore page

This PR has APIs and UI components for explore page.

- Files to review:
   - [ ] src/pages/Explore.jsx
   - [ ] src/reducers/posts.reducer.js
   - [ ] src/contexts/posts.context.js
   - [ ] src/components/Reusable/FeedPageHOC.jsx
   - [ ] src/components/Reusable/PostsHOC.jsx

- Use this link for preview -  https://deploy-preview-4--remarkable-flan-74a807.netlify.app/explore

v-1

Thanks.

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "baseUrl": "src"
+  },
+  "exclude": ["node_modules", "build"],
+  "include": ["src"]
+}
```
Comment (Developers-codz) [inline jsconfig.json:7]: Appreciation- used relative path instead of absolute path

File: src/components/Explore/ExplorePost.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import styles from './explore.module.css'
+
+const ExplorePost = ({ post: { _id, username, content, likes: { likeCount }, createdAt } }) => {
+    const { isUserLoggedIn } = useAuth()
+    const { likePost, dislikePost, bookmarkPost, postsDispatch } = usePosts()
+
+    const handlePostBookmark = async () => {
+        if (isUserLoggedIn) {
+            const response = await bookmarkPost(_id)
+            console.log(response)
```
Comment (Developers-codz) [inline src/components/Explore/ExplorePost.jsx]: can remove console if not needed further

File: src/components/Explore/ExplorePost.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                <p className='txt-secondary txt-md txt-300'>{`likes ${likeCount}`}</p>
+
+                <div className='flx'>
+
+                    <button onClick={handlePostBookmark} className='btn-txt txt-md txt-secondary txt-300 mg-right-xs'>
+                        {/* {isPostBookmarked ? 'bookmarked' : 'bookmark'} */}
+                        bookmark
+                    </button>
+
+                    {/* {isPostLiked ? 'liked' : 'like'} */}
```
Comment (Developers-codz) [inline src/components/Explore/ExplorePost.jsx:71]: remove commented code

File: src/components/Home/HomeMain.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    return (
+        <main className='flx flx-column pd-md'>
+
+            <p className='txt-secondary txt-xxlg txt-500'>sneaker culture</p>
+
+            <p className='txt-secondary txt-xlg txt-400 mg-top-xs mg-btm-xxlg'>join the gang</p>
+
+            <button onClick={navigateToSignup} className={`btn-solid ${styles.btnSignup} bg-primary txt-primary txt-lg txt-400 pd-top-s pd-btm-s brd-s mg-top-md mg-btm-md`}>sign up</button>
```
Comment (Developers-codz) [inline src/components/Home/HomeMain.jsx]: so many classname I think wrapping all css property in one class would be better ?? `btn-primary` or `btn-otline-primary` can be given

**Review by Developers-codz — APPROVED** [review]
Good work on code, Everything Looks Good to merge
Suggested few changes

File: src/components/Explore/ExplorePost.jsx (hunk 2 above)
Comment (foolhardy21) [reply src/components/Explore/ExplorePost.jsx]: Thanks for pointing it out.

File: src/components/Home/HomeMain.jsx (hunk 4 above)
Comment (foolhardy21) [reply src/components/Home/HomeMain.jsx]: Sure, I'll improve this.

File: src/components/Reusable/FeedPageHOC.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                className={`bg-off-secondary ${styles.feedPageGrid}`}>
+
+                <div className={`${styles.navDiv} flx flx-maj-end`}>
+                    <NavBar />
+                </div>
+
+                <PostsSection />
+
+                <div className={styles.extraDiv}>
+                    {/* third div empty for now. for searchbar and trending */}
```
Comment (anik31) [inline src/components/Reusable/FeedPageHOC.jsx:19]: appreciation - layout is good

**Review by anik31 — APPROVED** [review]
Nice work, everything looks good

**Pull request merged**